ktorm-core/src/main/kotlin/me/liuwj/ktorm/expression/SqlFormatter.kt

```diff
@@ -73,14 +73,43 @@ open class SqlFormatter(
         }
     }
 
-    open protected val String.quoted: String get() {
-        if (this.toUpperCase() in database.keywords || !this.isIdentifier) {
-            return "${database.identifierQuoteString}${this}${database.identifierQuoteString}".trim()
+    protected val String.quoted: String get() {
+        val shouldQuote = database.alwaysQuoteIdentifiers
+            || !this.isIdentifier
+            || this.toUpperCase() in database.keywords
+            || this.isMixedCase && !database.supportsMixedCaseIdentifiers && database.supportsMixedCaseQuotedIdentifiers
```

Comment:
@lyndsysimon We will quote the identifier when it meet any of these conditions: 

- The `alwaysQuoteIdentifiers` param is set to `true`.
- The identifier is not valid.
- The identifier is a SQL keyword. 
- The identifier is mixed case and the database doesn't support mixed case for unquoted identifiers but supports quoted ones.